appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/AbstractSessionContextImpl.java

```diff
@@ -85,15 +81,15 @@ public void setInstanceKey(Object instanceKey) {
 
     @Override
     public String toString() {
-        return ejbName + "; id: " + instanceKey;
+        return ejbName + "; id: " + getInstanceKey();
```

Comment:
It is weird, but instance key has overridable setter and getter. To be refactored later.

Reply:
Actually it does not matter in real time, but it is an abstract class and it should expect that this method would be overriden.
This is very aften case when I always have to override also toString method whenever I override any getter. Generally it depends on the whole concept, ie. equals can use getters; if it doesn't toString should not do it too to avoid confusion. But even this is not a law, sometimes toString prints only an id, sometimes classname and hashcode ...
Also - the old version - if anyone would override the getter, parent would still use the field and child would use different value.

We could also make getInstanceKey final, but ... it is an Object, it has both setter and getter ... I think this is better.